testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeCompatibility.java

```diff
@@ -0,0 +1,129 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.launcher.env.environment;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tests.product.launcher.docker.DockerFiles;
+import io.trino.tests.product.launcher.env.DockerContainer;
+import io.trino.tests.product.launcher.env.Environment;
+import io.trino.tests.product.launcher.env.EnvironmentProvider;
+import io.trino.tests.product.launcher.env.common.Hadoop;
+import io.trino.tests.product.launcher.env.common.Standard;
+import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import javax.inject.Inject;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.wait.strategy.Wait.forHealthcheck;
+import static org.testcontainers.containers.wait.strategy.Wait.forLogMessage;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class SinglenodeCompatibility
+        extends EnvironmentProvider
+{
+    private static final int SERVER_PORT = 18080;
+    private static final String COMPATIBILTY_TEST_CONTAINER_NAME = "compatibility-test-coordinator";
+
+    private final DockerFiles dockerFiles;
+    private final DockerFiles.ResourceProvider configDir;
+    private final PortBinder portBinder;
+
+    private Config extraConfig;
+
+    @Inject
+    public SinglenodeCompatibility(Standard standard, Hadoop hadoop, DockerFiles dockerFiles, PortBinder portBinder)
+    {
+        super(ImmutableList.of(standard, hadoop));
+        this.dockerFiles = dockerFiles;
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/singlenode-compatibility");
+        this.portBinder = portBinder;
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        configureCompatibilityTestContainer(builder, extraConfig);
+    }
+
+    private void configureCompatibilityTestContainer(Environment.Builder builder, Config config)
+    {
+        String configDir = getConfigurationDirectory(config.getCompatibilityTestDockerImage());
+        DockerContainer container = new DockerContainer(config.getCompatibilityTestDockerImage(), COMPATIBILTY_TEST_CONTAINER_NAME)
+                .withNetworkAliases(COMPATIBILTY_TEST_CONTAINER_NAME + ".docker.cluster")
```

Comment:
redundant?

Reply:
I'm wasn't sure. I "copied" this. Is this used only for the SSL certs? If so then I can remove this.

Reply:
Yes, that's for the SSL certificates (or keytabs)